src/MainWindow.cc

```diff
@@ -470,6 +477,12 @@ bool WindowConfig::MergeFromXML(const std::string &_windowXml)
     }
   }
 
+  // Show dialog on exit
+  if (auto dialogOnExitElem = winElem->FirstChildElement("dialog_on_exit"))
```

Comment:
In my opinion, I think the confirmation dialog should always appear by default (regardless if `<dialog_on_exit>true</dialog_on_exit>` is in `gui.config` or not) and when the user does not want the confirmation then they would set `<dialog_on_exit>false</dialog_on_exit>` but lets see what others think

cc @mabelzhang @ahcorde

Reply:
I don't have a strong opinion either way. I think it is safer for it to be on by default, but then I checked Gazebo-classic, and it doesn't seem to have a confirmation dialog. So... :woman_shrugging:

Reply:
In general I would to stick with the current behaviour and document this new feature. I open and close many times ign-gazebo and this is adding an overhead for me.

Maybe we can discuss in the next ignition meeting

Reply:
I think that leaving this off by default is the safer option. 
Although `ignition-gazebo` is the direct consumer of `ignition-gui` we probably don't want to change the behavior of all the applications that are developed on top of `ignition::gui::Application` (extern projects).